frontend/src/root.jsx

```diff
@@ -31,38 +36,57 @@ import AddActionPlan from "./modules/action-plan/view";
 import AddTechnology from "./modules/technology/view";
 import AddPolicy from "./modules/policy/view";
 
-api.get("/tag").then((resp) => {
-  UIStore.update((e) => {
-    e.tags = resp.data;
-  });
-});
-api.get("/currency").then((resp) => {
-  UIStore.update((e) => {
-    e.currencies = resp.data;
-  });
-});
-api.get("/country").then((resp) => {
-  UIStore.update((e) => {
-    e.countries = uniqBy(resp.data).sort((a, b) =>
-      a.name.localeCompare(b.name)
-    );
-  });
-});
-api.get("/country-group").then((resp) => {
-  UIStore.update((e) => {
-    e.regionOptions = resp.data.filter((x) => x.type === "region");
-  });
-  UIStore.update((e) => {
-    e.meaOptions = resp.data.filter((x) => x.type === "mea");
-  });
-});
-api.get("/organisation").then((resp) => {
-  UIStore.update((e) => {
-    e.organisations = uniqBy(sortBy(resp.data, ["name"])).sort((a, b) =>
-      a.name.localeCompare(b.name)
-    );
+api
```

Comment:
I'm curious why this refactored code is better than what we had before. Could you this refactor?

Reply:
I found on another project, the `useEffect` function running for each pullstate updated, so I decided to make it one time updated after all API call.